hyperspy/io_plugins/digital_micrograph.py

```diff
@@ -806,14 +806,17 @@ def _get_date(self, date):
             _logger.warning("Date string, %s,  could not be parsed", date)
 
     def _get_microscope_name(self, ImageTags):
-        try:
-            if ImageTags.Session_Info.Microscope != "[]":
-                return ImageTags.Session_Info.Microscope
-        except AttributeError:
-            if 'Name' in ImageTags['Microscope_Info'].keys():
-                return ImageTags.Microscope_Info.Name
-            elif 'Microscope' in ImageTags['Microscope_Info'].keys():
-                return ImageTags.Microscope_Info.Microscope
+        locations = (
+            "Session_Info.Microscope",
+            "Microscope_Info.Name",
+            "Microscope_Info.Microscope",
+        )
+        for loc in locations:
+            mic = ImageTags.get_item(loc)
+            if mic not in (None, "[]"):
```

Comment:
This looks much better now.
I think that `if mic is not None and len(mic) > 0:` would be better to check that this is not an empty string or an empty list. This `"[]"` looks wrong to me, I have checked a few files (from different DM version) and I can't find any having.

Reply:
You're right, I forgot the empty string.

Regarding "[]", there must be a reason for it. It would have been better to add a comment explaining the purpose of `"[]"`, but, as the chances of this becoming a microscope name are remote, leaving as is is harmless. I suggest to leave it just in case a particular DM version / DM setup marked this field as empty using [].

Reply:
Yes and it doesn't hear much anyway!
I have a file where I am getting the following (I possibly empty manually the tag in DM):
`└── microscope = []`
and adding an empty list in your tuple should fix it.

Reply:
Merge it then?

Reply:
Sorry, I havenn't been very clear. To avoid the unwanted behaviour mentioned above `└── microscope = []`, it needs to be changed to:
```python
if mic not in (None, "[]", "", []):
```